libraries/LittleFS/src/LittleFS.h

```diff
@@ -592,6 +598,26 @@ class LittleFSDirImpl : public DirImpl
         return _dir.get();
     }
 
+    uint32_t _getAttr4(char attr) {
+        if (!_valid) {
+            return 0;
+        }
+        int nameLen = 3; // Slashes, terminator
+        nameLen += _dirPath.get() ? strlen(_dirPath.get()) : 0;
+        nameLen += strlen(_dirent.name);
+        char *tmpName = (char*)malloc(nameLen);
```

Comment:
While here, I suggest replacing this with new[] into a unique_ptr and dropping the free below

Reply:
Practically the MAXPATH should be <128 bytes.  I should really just make this a stack var with a dynamic size.  C++ is cool that way. :)

Reply:
These are now stack temps in the latest push.